src/main/java/org/hoverla/bibernate/connectionpool/util/DriverDataSource.java

```diff
@@ -0,0 +1,145 @@
+    private final Properties driverProperties;
+    private Driver driver;
+
+    public DriverDataSource(String jdbcUrl, String driverClassName, String username, String password) {
+        this.jdbcUrl = jdbcUrl;
+        this.driverProperties = new Properties();
+
+        if (username != null) {
+            driverProperties.put(USER, driverProperties.getProperty(USER, username));
+        }
+        if (password != null) {
+            driverProperties.put(PASSWORD, driverProperties.getProperty(PASSWORD, password));
+        }
+
+        if (driverClassName != null) {
+            var drivers = DriverManager.getDrivers();
+            while (drivers.hasMoreElements()) {
+                var d = drivers.nextElement();
+                if (d.getClass().getName().equals(driverClassName)) {
+                    driver = d;
+                    break;
+                }
+            }
+
+            if (driver == null) {
+                log.warn("Registered driver with driverClassName={} was not found, trying direct instantiation.", driverClassName);
+                Class<?> driverClass = null;
+                var threadContextClassLoader = Thread.currentThread().getContextClassLoader();
+                try {
+                    if (threadContextClassLoader != null) {
+                        try {
+                            driverClass = threadContextClassLoader.loadClass(driverClassName);
+                            log.debug("Driver class {} found in Thread context class loader {}", driverClassName, threadContextClassLoader);
+                        } catch (ClassNotFoundException e) {
+                            log.debug("Driver class {} not found in Thread context class loader {}, trying classloader {}",
+                                    driverClassName, threadContextClassLoader, this.getClass().getClassLoader());
+                        }
+                    }
+
+                    if (driverClass == null) {
+                        driverClass = this.getClass().getClassLoader().loadClass(driverClassName);
+                        log.debug("Driver class {} found in the Configuration class classloader {}", driverClassName, this.getClass().getClassLoader());
+                    }
+                } catch (ClassNotFoundException e) {
+                    log.debug("Failed to load driver class {} from Configuration class classloader {}", driverClassName, this.getClass().getClassLoader());
+                }
+
+                if (driverClass != null) {
+                    try {
+                        driver = (Driver) driverClass.getDeclaredConstructor().newInstance();
+                    } catch (Exception e) {
+                        log.warn("Failed to create instance of driver class {}, trying jdbcUrl resolution", driverClassName, e);
+                    }
+                }
+            }
+        }
+
+        final var sanitizedUrl = jdbcUrl.replaceAll("([?&;]password=)[^&#;]*(.*)", "$1<masked>$2");
```

Comment:
Please move a regex to constants, or at least make it as a variable in this class